forcedotcom/dev-scripts
# chore: bump typedoc to 0.22.6

### What does this PR do?
Bumps `typedoc` to v0.22.6

BREAKING CHANGE: typedoc v0.20.0 introduced a lot of breaking changes,
see https://github.com/TypeStrong/typedoc/releases/tag/v0.20.0

**DEPENDS ON: https://github.com/forcedotcom/dev-config/pull/2**

**QA**:

1. Clone sdr: `git@github.com:forcedotcom/source-deploy-retrieve.git`
2. Checkout this PR 
3. Checkout this [PR](https://github.com/forcedotcom/dev-config/pull/2) and `yarn link` it into `dev-scripts`
4. Specify `dev-scripts` as a local package in sdr's package.json: `"@salesforce/dev-scripts": "../dev-scripts"`
5. in sdr, run `yarn install`
6. In sdr, remove the contents of `typedoc.js` and paste this:
```js
{
    "theme": "default",
    "excludePrivate": true,
    "excludeProtected": true,
    "entryPoints": "src"
}

```
7. Run `yarn docs` and see that docs are generated (you can open `./docs/index.html` in your browser).

**Roll out**:
This has to be done manually, typedoc now requires an entrypoint, every repo using `dev-scripts` should have a `typedoc.js` file to specify the entrypoint(s).
There are some repos that already have a `typedoc.js` file(sfdx-core, sdr) , those should be updated to work with newer typedoc.

### What issues does this PR fix or reference?
@W-10159876@
